CRM/Extension/QueueDownloader.php

```diff
@@ -133,11 +135,67 @@ public function fillQueue(CRM_Queue_Queue $queue, array $downloads, bool $upgrad
       ['weight' => 300]
     );
 
-    if ($upgradeDb) {
-      $queue->createItem(
-        static::subtask(ts('Upgrade database'), 'upgradeDb'),
-        ['weight' => 400]
-      );
+    if ($autoApply) {
+      // We need to figure out the mix of activation-steps (enable/upgrade).
+      // If you have multiple (e.g. enable $X and also upgrade $Y), then... which runs first?
+      // In theory, there is no simple rule of ordering that will work for all imaginable scenarios.
+      // The main mitigating factor is that actual usage will be biased toward simple-cases.
+      // (Ex: The web UI only lets you add one extension at a time. The CLI allows multiple -- but user must explicitly choose each.)
+      // This implementation defers to the user -- applying extensions in the order requested.
+
+      $statuses = CRM_Extension_System::singleton()->getManager()->getStatuses();
+      $todos = [];
+      foreach (array_keys($downloads) as $key) {
+        switch ($statuses[$key] ?? CRM_Extension_Manager::STATUS_UNINSTALLED) {
+          case CRM_Extension_Manager::STATUS_UNINSTALLED:
+            $todos[] = ['enable', $key];
```

Comment:
@totten I presume enable covers install etc as well right?

Reply:
Yup. `CRM_Extension_Manager::enable()` and `CRM_Extension_Manager::install()` are literally the same thing.

(*For an ext-author implementing `hook_install`/`hook_enable`, there is a difference. But for someone submitting a request to APIv3 or `CRM_Extension_Manager`, they are the same thing -- if they were different, it would be error-prone b/c callers would have to reliably choose the right one.*)